aioesphomeapi/client.py

```diff
@@ -284,7 +256,7 @@ def on_msg(msg: message.Message) -> None:
             if isinstance(msg, CameraImageResponse):
                 data = image_stream.pop(msg.key, bytes()) + msg.data
                 if msg.done:
-                    on_state(CameraState(key=msg.key, image=data))
```

Comment:
The issue people are experiencing on HA 2021.7 is because the new `from_pb` function only matches the attributes in the dataclass by exact name. That is `image` in `CameraState` but in `CameraImageResponse` it is `data`. Simple fix is to rename the attr to data and the references inside HA